app/screens/create_or_edit_channel/create_or_edit_channel.tsx

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect, useReducer, useCallback, useMemo} from 'react';
+import {useIntl} from 'react-intl';
+import {
+    Keyboard,
+} from 'react-native';
+import {Navigation} from 'react-native-navigation';
+
+import {patchChannel as handlePatchChannel, createChannel, switchToChannelById} from '@actions/remote/channel';
+import CompassIcon from '@app/components/compass_icon';
+import ChannelInfoForm from '@app/screens/create_or_edit_channel/channel_info_form';
+import {General} from '@constants';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {dismissModal, setButtons} from '@screens/navigation';
+import {validateDisplayName} from '@utils/channel';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+import type ChannelInfoModel from '@typings/database/models/servers/channel_info';
+
+type Props = {
+    componentId: string;
+    channel?: ChannelModel;
+    channelInfo?: ChannelInfoModel;
+}
+
+const CLOSE_BUTTON_ID = 'close-channel';
+const EDIT_BUTTON_ID = 'update-channel';
+const CREATE_BUTTON_ID = 'create-channel';
+
+enum RequestActions {
+    START = 'Start',
+    COMPLETE = 'Complete',
+    FAILURE = 'Failure',
+}
+
+interface RequestState {
+    error: string;
+    saving: boolean;
+}
+
+interface RequestAction {
+    type: RequestActions;
+    error?: string;
+}
+
+const close = (componentId: string): void => {
+    Keyboard.dismiss();
+    dismissModal({componentId});
+};
+
+const isDirect = (channel?: ChannelModel): boolean => {
+    return channel?.type === General.DM_CHANNEL || channel?.type === General.GM_CHANNEL;
+};
+
+const CreateOrEditChannel = ({
+    componentId,
+    channel,
+    channelInfo,
+}: Props) => {
+    const intl = useIntl();
+    const {formatMessage} = intl;
+    const theme = useTheme();
+    const serverUrl = useServerUrl();
+
+    const editing = Boolean(channel);
+
+    const rightButton = useMemo(() => {
+        return {
+            testID: 'edit_channel.save.button',
+            id: editing ? EDIT_BUTTON_ID : CREATE_BUTTON_ID,
+            enabled: false,
+            showAsAction: 'always' as const,
+            color: theme.sidebarHeaderTextColor,
+            text: editing ? formatMessage({id: 'mobile.edit_channel', defaultMessage: 'Save'}) : formatMessage({id: 'mobile.create_channel', defaultMessage: 'Create'}),
+        };
```

Comment:
Use the `buildNavigationButton` function as it sets the font used in this types of buttons.. in fact you could even extend it to use the text as it seems that I forgot to add it.